### PR TITLE
feat: [#347] Switch logging channels on runtime

### PR DESCRIPTION
**Pull request by kkumar-gcc**
Closes https://github.com/goravel/goravel/issues/347

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: log/application.go [hunk 1]
```diff
@@ -13,33 +13,71 @@ import (
 type Application struct {
 	log.Writer
 	instance *logrus.Logger
+	config   config.Config
 }
 
 func NewApplication(config config.Config) *Application {
+	instance := newLogger(config, config.GetString("logging.default"))
```
Comment (hwbrzzl) [inline log/application.go:20]: Didn't call `registerHook`? There may be some errors and the ci failed.

File: log/application.go [hunk 2]
```diff
@@ -32,14 +32,56 @@ func NewApplication(config config.Config) *Application {
...
+			continue
+		}
+
+		channelPath := "logging.channels." + channel
+		driver := r.config.GetString(channelPath + ".driver")
+
+		if driver == log.StackDriver {
+			color.Red().Println("stack driver can't include self channel")
+			return nil
+		}
```
Comment (hwbrzzl) [inline log/application.go:78]: We may don't need to add this limitation, `registerHook` will handle the logic.

**Review by hwbrzzl — COMMENTED** [review]
Good, please optimize test cases.

File: log/application.go (hunk 2 above)
Comment (kkumar-gcc) [reply log/application.go:78]: Naah registerHook won't handle it.

**Review by devhaozi — APPROVED** [review]
LGTM

File: log/application.go (hunk 2 above)
Comment (hwbrzzl) [reply log/application.go:78]: The stack function doesn't add channels into the config? Do you mean  this?

image

The method `facades.Log().Stack([]string{"stack", "single"})` is allowed, you seem to be missing something?

Comment (kkumar-gcc) [reply log/application.go:78]: Oh, sorry, I was mistaking the channel for the driver.

File: log/application_test.go
Comment (hwbrzzl) [inline log/application_test.go:1]: Great

**Review by hwbrzzl — APPROVED** [review]

**Pull request merged**